### PR TITLE
Rep 2378 intra filter logging nulls

### PR DESCRIPTION
**Pull request by Mario-Lopez**
When trace logging is on, PowerFilterChain.java will log the request and response along with the filter information.  Filters have an optional ID which was getting logged as null along side the filter name (e.g. "null-ip-identify" for a filter named "ip-identity" with no ID).

I added code to not log the filter ID when it's null or an empty string.  ResponseLog.java and RequestLog.java were updated to just use the filter name when the ID is empty (i.e. null or an empty string).

These classes didn't have unit tests, so I added new classes to unit test the updates I made (using Scala Test).  I also updated the functional test (IntrafilterLoggingTest.groovy which already had two filters configured) to make one of the filters have an ID and the other one not have an ID.  The test verifies that "context_1-ip-identity" and "client-auth" (no ID) gets logged respectively.  All unit tests pass, and the functional test that was updated passes as well.

### CODE REVIEW

**Comment by wdschei** [comment]
Ok to test.

**Comment by adrianjgeorge** [comment]
ok to test

File: repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/intrafilterLogging/RequestLog.java [hunk 1]
```diff
@@ -41,33 +44,58 @@
...
+     * @param filterContext {@link FilterContext}
+     * @throws IOException if there's an issue converting the response body to a string
+     */
     public RequestLog(MutableHttpServletRequest mutableHttpServletRequest,
                       FilterContext filterContext) throws IOException {
 
         preamble = "Intrafilter Request Log";
         timestamp = new DateTime().toString();
-        currentFilter = filterContext.getFilterConfig().getId() + "-" + filterContext.getFilterConfig().getName();
+        currentFilter = getFilterDescription(filterContext.getFilterConfig());
```
Comment (adrianjgeorge) [inline repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/intrafilterLogging/RequestLog.java]: you shouldn't call instance methods in a constructor behavior can be erratic. Either make it a static method or better yet just do a simple ternary.

File: repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/intrafilterLogging/ResponseLog.java [hunk 2]
```diff
@@ -39,12 +45,18 @@
...
+     * @param filterContext {@link FilterContext}
+     * @throws IOException if there's an issue converting the response body to a string
+     */
     public ResponseLog(MutableHttpServletResponse mutableHttpServletResponse,
                        FilterContext filterContext) throws IOException {
 
         preamble = "Intrafilter Response Log";
         timestamp = new DateTime().toString();
-        currentFilter = filterContext.getFilterConfig().getId() + "-" + filterContext.getFilterConfig().getName();
+        currentFilter = getFilterDescription(filterContext.getFilterConfig());
```
Comment (adrianjgeorge) [inline repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/intrafilterLogging/ResponseLog.java]: same here

**Comment by adrianjgeorge** [comment]
You added a lot of javadocs that don't really provide any value, because they basically say the exact same thing the method and variable names do. In general we don't add javadocs unless something is none obvious about usage.

**Comment by Mario-Lopez** [comment]
I'll update it to a ternary op.  That JavaDoc, tho...  Most of the JavaDocs I added were for code that took me more than a minute to reason about.  I removed the docs for one of the methods.  Will the rest of the JavaDocs keep you up at night if I leave them in?

**Comment by Mario-Lopez** [comment]
retest this please

**Comment by Mario-Lopez** [comment]
retest this please

File: repose-aggregator/core/core-lib/src/test/scala/org/openrepose/powerfilter/intrafilterLogging/RequestLogTest.scala [hunk 3]
```diff
@@ -0,0 +1,101 @@
+package org.openrepose.powerfilter.intrafilterLogging
```
Comment (wdschei) [inline repose-aggregator/core/core-lib/src/test/scala/org/openrepose/powerfilter/intrafilterLogging/RequestLogTest.scala]: This file needs the License header to run on Jenkins.

File: repose-aggregator/core/core-lib/src/test/scala/org/openrepose/powerfilter/intrafilterLogging/ResponseLogTest.scala [hunk 4]
```diff
@@ -0,0 +1,97 @@
+package org.openrepose.powerfilter.intrafilterLogging
```
Comment (wdschei) [inline repose-aggregator/core/core-lib/src/test/scala/org/openrepose/powerfilter/intrafilterLogging/ResponseLogTest.scala]: This file needs the License header to run on Jenkins.

File: repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/core/intrafilterlogging/IntraFilterLoggingTest.groovy [hunk 5]
```diff
@@ -0,0 +1,62 @@
+package features.core.intrafilterlogging
```
Comment (wdschei) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/core/intrafilterlogging/IntraFilterLoggingTest.groovy]: This file needs the License header to run on Jenkins.

**Comment by dmnjohns** [comment]
retest this please

File: repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/intrafilterLogging/RequestLog.java [hunk 6]
```diff
@@ -41,17 +44,25 @@
     String requestBody;
     HashMap<String, String> headers;
 
+    /**
+     * Constructor populates all of the fields necessary for logging.
```
Comment (adrianjgeorge) [inline repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/intrafilterLogging/RequestLog.java]: It's not going to keep me up at night, but what value do you see in this? it tells me absolutely nothing that wasn't in already communicated by the names of the parameters

Comment (Mario-Lopez) [inline repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/intrafilterLogging/RequestLog.java]: There are a few things perplexing about this class.  I usually don't see package-level visibility for class fields.  To me that implies that there is another class in that same package that may be populating these fields (especially considering there are no setter methods).  I looked.  There is not.  Either it was a mistake or the serialization code requires it for some reason.  In any case, that particular JavaDoc addresses that.  The next time I have to go through this code, I don't have to worry about what else might be populating the fields.  I will know that at some point someone decided to tightly couple this POJO with MutableHttpServletRequest and FilterContext for some reason, and this JavaDoc allows me to remember this instantly.

File: repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/intrafilterLogging/RequestLog.java [hunk 7]
```diff
@@ -20,17 +20,20 @@
...
 import javax.servlet.ServletInputStream;
 import java.io.IOException;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.LinkedHashMap;
-import java.util.List;
+import java.util.*;
 
+/**
+ * POJO that is used to log details about a response when the log level is set to TRACE.
```
Comment (adrianjgeorge) [inline repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/intrafilterLogging/RequestLog.java]: You've created an inaccuracy here. the behavior of this class doesn't care about the logging level, that's only relevant in where it's called.

Comment (Mario-Lopez) [inline repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/intrafilterLogging/RequestLog.java]: Potential inaccuracy*

Considering the constructor contains the logic for what gets logged (i.e. what goes into the fields), I thought the expected log level was relevant here.  I can change "when the log level is set to TRACE" to "(may not always be logged due to log level)" in order to decrease the likeliness of the documentation becoming stale.

Comment (Mario-Lopez) [inline repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/intrafilterLogging/RequestLog.java]: My apologies.  No documentation is better than poor documentation.  I'll push an update.

File: repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/intrafilterLogging/RequestLog.java (hunk 6 above)
Comment (Mario-Lopez) [inline repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/intrafilterLogging/RequestLog.java]: My apologies.  No documentation is better than poor documentation.  I'll push an update.

**Comment by adrianjgeorge** [comment]
you removed the line comments? those seemed to be the more valuable ones

**Pull request merged**